hogvm/typescript/src/__tests__/bytecode.test.ts

```diff
@@ -1,168 +1,347 @@
-import { executeHogQLBytecode, Operation as op } from '../bytecode'
+import { exec, Operation as op } from '../bytecode'
+
+export function delay(ms: number): Promise<void> {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms)
+    })
+}
 
 describe('HogQL Bytecode', () => {
     test('execution results', async () => {
         const fields = { properties: { foo: 'bar', nullValue: null } }
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 2, op.INTEGER, 1, op.PLUS], fields)).toBe(3)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 2, op.INTEGER, 1, op.MINUS], fields)).toBe(-1)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 2, op.INTEGER, 3, op.MULTIPLY], fields)).toBe(6)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 2, op.INTEGER, 3, op.DIVIDE], fields)).toBe(1.5)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 2, op.INTEGER, 3, op.MOD], fields)).toBe(1)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 2, op.INTEGER, 1, op.AND, 2], fields)).toBe(true)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 0, op.INTEGER, 1, op.OR, 2], fields)).toBe(true)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 0, op.INTEGER, 1, op.AND, 2], fields)).toBe(false)
-        expect(await executeHogQLBytecode(['_h', op.INTEGER, 1, op.INTEGER, 0, op.INTEGER, 1, op.OR, 3], fields)).toBe(
-            true
+        expect(await exec(['_h'], fields)).toBe(null)
```

Comment:
Long term - we should make this run both python and nodejs in parallel against the same test cases if possible

Reply:
Absolutely. Especially as we're talking about adding a third target, Rust, as well.